app/src/main/java/study_dev/testbed/studyhopper/ui/dashboard/DashboardFragment.java

```diff
@@ -47,35 +52,65 @@
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
 
-
-//        final ArrayList<studyGroupItem> studyGroupList = new ArrayList<>();
-//        studyGroupList.add(new studyGroupItem(R.drawable.ic_group_color_purple, "Mobile Devices", "COP 4656"));
-//        studyGroupList.add(new studyGroupItem(R.drawable.ic_group_color_blue, "Automata Fun", "COT 4210"));
-//        studyGroupList.add(new studyGroupItem(R.drawable.ic_group_color_green, "How to not go to Jail", "CIS 4250"));
-//        studyGroupList.add(new studyGroupItem(R.drawable.ic_group_color_purple, "Hendrix Fun", "COP 4970"));
-//        studyGroupList.add(new studyGroupItem(R.drawable.ic_group_color_blue, "Hadoop & Big Data", "CIS 4930"));
-//        studyGroupList.add(new studyGroupItem(R.drawable.ic_group_color_green, "GRE Prep", "TestPrep"));
-//        studyGroupList.add(new studyGroupItem(R.drawable.ic_group_color_blue, "Test", "Test"));
-
-
         View v = inflater.inflate(R.layout.fragment_dashboard, container, false);
 
         welcomeMsg = v.getRootView().findViewById(R.id.welcome_txt);
 
         groupFinderCard = v.getRootView().findViewById(R.id.studyGroupFinderCard);
+        recyclerView = v.getRootView().findViewById(R.id.study_group_recycler_view);
 
+        // Retrieve user's email address from FirebaseAuth
+        FirebaseUser user = mAuth.getCurrentUser();
+        if(user != null){
+            userEmail = user.getEmail();
+            Toast.makeText(getContext(), userEmail, Toast.LENGTH_SHORT).show();
+        }
+
+        Query userQuery = userRef.whereEqualTo("email", userEmail);
+
+        userQuery.get().addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                    @Override
+                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                        if(task.isSuccessful()) {
+                            for(QueryDocumentSnapshot document : task.getResult()) {
+                                userDocumentId = document.getId();
+                                groupRef = db.collection("users").document(userDocumentId)
+                                        .collection("groups");
+                                buildWelcomeMsg();
+
+                                setUpRecyclerView();
+
+                                adapter.setOnItemClickListener(new GroupAdapter.OnItemClickListener() {
+                                    @Override
+                                    public void onItemClick(DocumentSnapshot documentSnapshot, int position) {
+                                        String id = documentSnapshot.getId();
+                                        Toast.makeText(getContext(),
+                                                "Position: " + position + " ID: " + id, Toast.LENGTH_SHORT).show();
+                                        Activity activity = (Activity) getContext();
```

Comment:
what is this doing? is it restarting the current activity?

Reply:
I have overrode the onClick listener in the GroupAdapter.  However, because we are in a fragment I still wanted to override the pending transition fragment from the Dashboard.java activity. Essentially I am digging into the root to do this.